src/state.c

```diff
@@ -1138,7 +1138,14 @@ xkb_state_mod_index_is_active(struct xkb_state *state,
     if (idx >= xkb_keymap_num_mods(state->keymap))
         return -1;
 
-    return !!(xkb_state_serialize_mods(state, type) & (1u << idx));
+    if (is_real_mod(state->keymap, idx)) {
+        return !!(xkb_state_serialize_mods(state, type) & (1u << idx));
+    } else {
+        /* WARNING: this may over matches */
+        return !!((xkb_state_serialize_mods(state, type) &
+                   state->keymap->mods.mods[idx].mapping) ==
+                  state->keymap->mods.mods[idx].mapping);
```

Comment:
at the risk of sounding stupid: can we do this with a temp variable please - with the linebreaks and parens it gets a bit hard to parse:
```c
   something_t mapping = state->keymap->mods[idx].mapping;
   return !!((xkb_state_serial_mods(state, type) & mapping) == mapping);
```

Reply:
Done